src/checkout.c

```diff
@@ -2521,7 +2521,8 @@ int git_checkout_iterator(
 
 	if (data.opts.baseline_index) {
 		if ((error = git_iterator_for_index(
-				&baseline, data.opts.baseline_index, &baseline_opts)) < 0)
+				&baseline, git_index_owner(data.opts.baseline_index),
```

Comment:
Imma hit up some refactoring of iterators, so I didn't want to bikeshed too hard on this yet.  That's a good idea though that I'll keep in mind for working on this.